src/sage/rings/laurent_series_ring_element.pyx

```diff
@@ -523,22 +523,32 @@ cdef class LaurentSeries(AlgebraElement):
             sage: f = -5/t^(10) + 1/3 + t + t^2 - 10/3*t^3 + O(t^5); f
             -5*t^-10 + 1/3 + t + t^2 - 10/3*t^3 + O(t^5)
 
-        Slicing is deprecated::
+        Slicing can be used to truncate, keeping the same precision::
 
-            sage: f[-10:2]
-            doctest:...: DeprecationWarning: polynomial slicing with a start index is deprecated, use list() and slice the resulting list instead
-            See https://github.com/sagemath/sage/issues/18940 for details.
+            sage: f[:2]
             -5*t^-10 + 1/3 + t + O(t^5)
+
+        Any other kind of slicing is an error, see :trac:`18940`::
+
+            sage: f[-10:2:2]
+            Traceback (most recent call last):
+            ...
+            NotImplementedError: polynomial slicing with a step is not defined
```

Comment:
Shouldn't this be `IndexError` or something like this?

Reply:
I read `NotImplementedError` as an invitation to implement this feature.

But here a feature has been deliberately removed.